src/neptune_api/client.py

```diff
@@ -334,55 +337,53 @@ def __init__(
         self,
         credentials: Credentials,
         client_id: str,
-        token_endpoint: str,
-        token_factory: Callable[[Client, Credentials], OAuthToken],
+        token_refreshing_endpoint: str,
+        api_key_exchange_factory: Callable[[Client, Credentials], OAuthToken],
         client: Client,
     ):
         self._credentials: Credentials = credentials
         self._client_id: str = client_id
-        self._token_endpoint: str = token_endpoint
-        self._token_factory: Callable[[Client, Credentials], OAuthToken] = token_factory
+        self._token_refreshing_endpoint: str = token_refreshing_endpoint
+        self._api_key_exchange_factory: Callable[[Client, Credentials], OAuthToken] = api_key_exchange_factory
 
         self._client = client
         self._token: Optional[OAuthToken] = None
 
-        print("Initialized")
-
     def _refresh_existing_token(self) -> None:
         if self._token is None:
-            # TODO: Better error handling
-            raise ValueError("Token is not set")
-
-        response = self._client.get_httpx_client().post(
-            url=self._token_endpoint,
-            data={
-                "grant_type": "refresh_token",
-                "refresh_token": self._token.refresh_token,
-                "client_id": self._client_id,
-                "expires_in": self._token.seconds_left,
-            },
-        )
-        # TODO: Error handling
-        data = response.json()
+            # This should never happen, but just in case
```

Comment:
if this should never happen maybe lets log it

Reply:
Sure, saved for later as we have to figure out how to introduce logger properly that could be controlled via main package.